src/main/java/org/commcare/formplayer/services/MenuSessionRunnerService.java

```diff
@@ -475,8 +475,10 @@ private void doPostAndSync(MenuSession menuSession, FormplayerSyncScreen screen)
         if (shouldSync) {
             String moduleName = ScreenUtils.getBestTitle(menuSession.getSessionWrapper());
             Map<String, String> extraTags = new HashMap<>();
-            extraTags.put(Constants.MODULE_NAME_TAG, moduleName);
+            Map<String, String> requestScopedTagNameAndValueMap = datadog.getRequestScopedTagNameAndValue();
 
+            requestScopedTagNameAndValueMap.computeIfPresent(Constants.REQUEST_INCLUDES_TAG, extraTags::put);
```

Comment:
we can probably make this logic to consolidate tags part of `recordCategoryTiming` method itself ?

Reply:
I think to reduce the number of tag combinations (thinking about cost) we wouldn't always want to include the requestScopedTags into the category timing metric. So in that case, we would decide which of the requestScopedTags we're interested in on the higher level and pass it into `recordCategoryTiming`. But let me know if i'm misunderstanding.

Reply:
mmm that could work. But the `requestScopedTag` would contain more than just the `REQUEST_INCLUDES_TAG` so are you imagining a flag for each tag? Like a parameter to `recordCategoryTiming ` for `add_request_includes_tag`, `add_module_name_tag`?

Reply:
no I was just imagining a single flag like ` includeRequestScopedTags` for `recordCategoryTiming`  that adds the request scoped tags to `extraTags` .. I think the main thing I am trying to avoid is duplication of the tags merging logic. there are 2 places in the PR where we are doing this. Another easier method might just be to have a method like `datadog.consolidateTags(extraTags)` that returns you the complete set of tags including the scoped tags.

Reply:
Ah that would relate back to the initial concern with tag combinations. `requestScopedTagNameAndValueMap` contains more than just the `REQUEST_INCLUDES_TAG` tag. And for costs, we won't want to include all the tags.

I am merging this PR for to make into the deploy, but happy to continue discussing this.